optimus/io/jdbc.py

```diff
@@ -227,31 +244,35 @@ def execute(self, query, limit=None):
         """
 
         # play defensive with a select clause
-        if self.db_driver == "oracle":
-            alias = " t"
-        elif self.db_driver == "presto":
-            alias = ""
+        if self.db_driver == DriverResolver.ORACLE.__str__():
+            query = "(" + query + ") t"
+        elif self.db_driver == DriverResolver.PRESTO.__str__():
+            query = "(" + query + ")"
+        elif self.db_driver == DriverResolver.CASSANDRA.__str__():
+            query = query
         else:
-            alias = " AS t"
-
-        query = "(" + query + self._limit(limit) + ")" + alias
+            query = "(" + query + ") AS t"
 
         logger.print(query)
         logger.print(self.url)
 
         conf = Spark.instance.spark.read \
-            .format("jdbc") \
+            .format("jdbc" if not DriverResolver.CASSANDRA.__str__() else "org.apache.spark.sql.cassandra") \
             .option("url", self.url) \
-            .option("dbtable", query) \
             .option("user", self.user)
 
-        if self.db_driver != "presto" and self.password is not None:
+        if self.db_driver != DriverResolver.PRESTO.__str__() and self.password is not None:
             conf.option("password", self.password)
 
-        if self.db_driver == "oracle" or self.db_driver == 'postgresql' or self.db_driver == 'presto':
+        if self.db_driver == DriverResolver.ORACLE.__str__() \
+                or self.db_driver == DriverResolver.POSTGRES_SQL.__str__() \
+                or self.db_driver == DriverResolver.PRESTO.__str__():
             conf.option("driver", self.driver_option)
 
-        return conf.load()
+        if self.db_driver == DriverResolver.CASSANDRA.__str__():
+            conf.options(table="main", keyspace=self.cassandra_keyspace)
+
+        return self._limit(conf.load(), limit)
```

Comment:
Any reason for adding the limit after loading the data instead of specifying in the query?

Reply:
@arpit1997 Yeah because the `cassandra-spark-connector` doesn't support query. So I don't have another way to enforce the limit other than doing it at the `df` level.